x-pack/plugins/actions/server/builtin_action_types/pagerduty.ts

```diff
@@ -260,12 +263,8 @@ function getBodyForEventAction(actionId: string, params: ActionParamsType): unkn
     summary: params.summary || 'No summary provided.',
     source: params.source || `Kibana Action ${actionId}`,
     severity: params.severity || 'info',
+    ...omitBy(pick(params, ['timestamp', 'component', 'group', 'class']), isUndefined),
   };
 
-  if (params.timestamp != null) data.payload.timestamp = params.timestamp;
-  if (params.component != null) data.payload.component = params.component;
-  if (params.group != null) data.payload.group = params.group;
-  if (params.class != null) data.payload.class = params.class;
-
```

Comment:
I made this change in the interest of making it easier to maintain longer term.
I spent a good few minutes tying to figure out why we were comparing to `null` a value that can only be `String` or `undefined`... I was convinced this was a bug... it took me _embarassingly_ long to notice the `!=` rather than `!==`.

To prevent others from going down the same rabbit hole, I've replaced this with an explicit check for `undefined`. :)

Reply:
This will change the payload so that previously params that were null were not included in the payload, but now they will, with a value of `null`.  If that's ok with the PagerDuty API, then it's fine with me :-)